dotnet-statsig/src/Statsig/Server/Evaluation/ConfigCondition.cs

```diff
@@ -29,19 +29,25 @@ internal ConfigCondition(string type, JToken targetValue, string op, string fiel
             IDType = idType;
         }
 
-        internal static ConfigCondition FromJObject(JObject jobj)
+        internal static ConfigCondition? FromJObject(JObject jobj)
         {
-            JToken type, targetValue, op, field, additionalValues, idType;
+            JToken? type, targetValue, op, field, additionalValues, idType;
             if (jobj == null || !jobj.TryGetValue("type", out type))
             {
                 return null;
             }
+            
+            var valuesMap = new Dictionary<string, object>();
+            if (jobj.TryGetValue("additionalValues", out additionalValues)) 
+            {
+                valuesMap = additionalValues.ToObject<Dictionary<string, object>>() ?? valuesMap;
+            }
             return new ConfigCondition(
-                type.Value<string>(),
+                type.Value<string>() ?? "",
                 jobj.TryGetValue("targetValue", out targetValue) ? targetValue : null,
                 jobj.TryGetValue("operator", out op) ? op.Value<string>() : null,
                 jobj.TryGetValue("field", out field) ? field.Value<string>() : null,
-                jobj.TryGetValue("additionalValues", out additionalValues) ? additionalValues.ToObject<Dictionary<string, object>>() : new Dictionary<string, object>(),
+                valuesMap,
```

Comment:
why the change here?

Reply:
Because ToObject could be null as well and it is awkward to write all that in a single statement.